### PR TITLE
chore: migrate repo to typescript

### PR DESCRIPTION
**Pull request by vinaybedre**

### CODE REVIEW

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ function Footer() {
       <GitHubButton
         href={ORG_URL}
         data-size="large"
-        data-show-count="true"
+        data-show-count={true}
```
Comment (sebastianvera) [inline src/components/Footer/Footer.tsx]: Weird that I copy and paste this from their website :P did typescript catch the error?

File: src/components/Header/Link/Link.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import { NavLink } from 'react-router-dom';
 
-function HeaderLink(props) {
+function HeaderLink(props: any) {
```
Comment (sebastianvera) [inline src/components/Header/Link/Link.tsx]: Should we add the correct types in a following PR?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (sebastianvera) [inline tsconfig.json]: I'll need to check some of this options, did you come up with this file yourself or was it generated? some of this configs might be useful for the BE repo too!

File: package.json [hunk 4]
```diff
@@ -8,12 +8,18 @@
     "url": "https://github.com/apipatth/ppoker-rest-react.git"
   },
   "dependencies": {
+    "@types/jest": "^24.0.23",
```
Comment (sebastianvera) [inline package.json]: should we add these as `devDependencies` instead?

File: package.json [hunk 5]
```diff
@@ -8,12 +8,18 @@
...
+    "@types/react-dom": "^16.9.4",
+    "@types/react-router-dom": "^5.1.2",
     "classnames": "^2.2.6",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
     "react-github-btn": "^1.1.1",
     "react-router-dom": "^5.1.2",
-    "react-scripts": "3.2.0"
+    "react-scripts": "3.2.0",
+    "typescript": "3.6.4"
```
Comment (sebastianvera) [inline package.json]: Any reason to avoid using typescript `3.7.X`? 🚀

File: src/components/Footer/Footer.tsx (hunk 1 above)
Comment (vinaybedre) [reply src/components/Footer/Footer.tsx]: Yes

File: src/components/Header/Link/Link.tsx (hunk 2 above)
Comment (vinaybedre) [reply src/components/Header/Link/Link.tsx]: A following pr sounds good

File: package.json (hunk 5 above)
Comment (vinaybedre) [reply package.json]: 3.7.x is with warnings. Safe to use 3.6.x

File: tsconfig.json (hunk 3 above)
Comment (vinaybedre) [reply tsconfig.json]: Auto generated

File: package.json (hunk 4 above)
Comment (vinaybedre) [reply package.json]: It's as per CRU. I can change to devDependency

Comment (sebastianvera) [reply package.json]: aaah I see, maybe it's to be able to build the app even when running `yarn install` with `NODE_ENV=production`, let's keep them as dependencies :)

**Pull request merged**